core/actions/incremental_table.ts

```diff
@@ -465,6 +466,30 @@ export class IncrementalTable extends ActionBuilder<dataform.Table> {
       VerifyProtoErrorBehaviour.SHOW_DOCS_LINK
     );
   }
+
+  private mapOnSchemaChange(onSchemaChange?: string|number) : dataform.OnSchemaChange {
```

Comment:
The type `number` is supported here, but I'm not sure what that would look like if you actually wanted to reference the named enum in the code (writing integers isn't really a CUJ we need to support).

So from that, just expecting string would make sense - we're not writing tests for the number version, and that would add more complexity.

However, I'm slightly confused why string works here in the first place. This step https://github.com/dataform-co/dataform/blob/c799ec73800f23d0cbf03e1862a4a841e79b1b6d/core/actions/incremental_table.ts#L462 checks that `onSchemaChange` is of the type specified in the proto, which I guess for enums includes strings.

So overall I think this implementation correct, but it would be good to do some manual testing for weird edge case behavior later

Reply:
I added comments to my mapping function which explains the logic:

The type of onSchemaChange depends on the source file:
- for sqlx it will have type "string"
- for action.yaml it will be converted to enum which is represented
in TypeScript as a "number".

It is a good idea to unify it during the verification case, will work on this in a separate PR.